### DISCUSSION TITLE
Can't remove DRM, don't know what I'm doing wrong

### DISCUSSION THREAD

**Discussion by Lolliponzo** [apprenticeharper/DeDRM_tools#2303] [Q&A] [answered]
I'm trying to convert a pdf but I always get this error. I have installed ADE 4.5.12.112 and I can read the pdf there.

calibre, version 7.6.0 (win32, embedded-python: True)
Errore di conversione: Non riuscita: Converti libro 1 di 1

Converti libro 1 di 1 
DeDRM v10.0.9: Trying to decrypt lnxc8wnj.pdf
DeDRM v10.0.9: lnxc8wnj.pdf is a PDF ebook with encryption EBX_HANDLER
DeDRM v10.0.9: lnxc8wnj.pdf is a PDF ebook (EBX) for UUID 034cb0cc-ea1a-4113-822f-474f8630b186
DeDRM v10.0.9: Looking for new default Adobe Digital Editions Keys after 0.1 seconds
DeDRM v10.0.9: Exception when getting default Adobe Key after 0.1 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 727, in PDFIneptDecrypt
  File "C:\Users\-----\AppData\Roaming\calibre\plugins\DeDRM.zip\adobekey.py", line 293, in adeptkeys
  File "C:\Users\-----\AppData\Roaming\calibre\plugins\DeDRM.zip\adobekey.py", line 273, in CryptUnprotectData
adobekey.ADEPTError: Failed to decrypt user key key (sic)
DeDRM v10.0.9: Finished after 0.2 seconds
Conversion options changed from defaults:
  verbose: 2
  pdf_serif_family: 'Tahoma'
  pdf_mono_family: 'Courier New'
  output_profile: 'generic_eink'
  read_metadata_from_opf: 'C:\\Users\\-----\\AppData\\Local\\Temp\\calibre_2fcid2td\\t9z0ksci.opf'
  pdf_sans_family: 'Tahoma'
Resolved conversion options
calibre version: 7.6.0
{'asciiize': False,
 'author_sort': None,
 'authors': None,
 'base_font_size': 0.0,
 'book_producer': None,
 'change_justification': 'original',
 'chapter': "//*[((name()='h1' or name()='h2') and re:test(., "
            "'\\s*((chapter|book|section|part)\\s+)|((prolog|prologue|epilogue)(\\s+|$))', "
            "'i')) or @class = 'chapter']",
 'chapter_mark': 'pagebreak',
 'comments': None,
 'cover': None,
 'custom_size': None,
 'debug_pipeline': None,
 'dehyphenate': True,
 'delete_blank_paragraphs': True,
 'disable_font_rescaling': False,
 'duplicate_links_in_toc': False,
 'embed_all_fonts': False,
 'embed_font_family': None,
 'enable_heuristics': False,
 'expand_css': False,
 'extra_css': None,
 'filter_css': '',
 'fix_indents': True,
 'font_size_mapping': None,
 'format_scene_breaks': True,
 'html_unwrap_factor': 0.4,
 'input_encoding': None,
 'input_profile': <calibre.customize.profiles.InputProfile object at 0x00000268FB7380D0>,
 'insert_blank_line': False,
 'insert_blank_line_size': 0.5,
 'insert_metadata': False,
 'isbn': None,
 'italicize_common_cases': True,
 'keep_ligatures': False,
 'language': None,
 'level1_toc': None,
 'level2_toc': None,
 'level3_toc': None,
 'line_height': 0.0,
 'linearize_tables': False,
 'margin_bottom': 5.0,
 'margin_left': 5.0,
 'margin_right': 5.0,
 'margin_top': 5.0,
 'markup_chapter_headings': True,
 'max_toc_links': 50,
 'minimum_line_height': 120.0,
 'new_pdf_engine': False,
 'no_chapters_in_toc': False,
 'no_images': False,
 'no_inline_navbars': False,
 'output_profile': <calibre.customize.profiles.GenericEink object at 0x00000268FB6E00D0>,
 'page_breaks_before': "//*[name()='h1' or name()='h2']",
 'paper_size': 'letter',
 'pdf_add_toc': False,
 'pdf_default_font_size': 20,
 'pdf_footer_template': None,
 'pdf_header_template': None,
 'pdf_hyphenate': False,
 'pdf_mark_links': False,
 'pdf_mono_family': 'Courier New',
 'pdf_mono_font_size': 16,
 'pdf_no_cover': False,
 'pdf_odd_even_offset': 0.0,
 'pdf_page_margin_bottom': 72.0,
 'pdf_page_margin_left': 72.0,
 'pdf_page_margin_right': 72.0,
 'pdf_page_margin_top': 72.0,
 'pdf_page_number_map': None,
 'pdf_page_numbers': False,
 'pdf_sans_family': 'Tahoma',
 'pdf_serif_family': 'Tahoma',
 'pdf_standard_font': 'serif',
 'pdf_use_document_margins': False,
 'prefer_metadata_cover': False,
 'preserve_cover_aspect_ratio': False,
 'pretty_print': False,
 'pubdate': None,
 'publisher': None,
 'rating': None,
 'read_metadata_from_opf': 'C:\\Users\\-----\\AppData\\Local\\Temp\\calibre_2fcid2td\\t9z0ksci.opf',
 'remove_fake_margins': True,
 'remove_first_image': False,
 'remove_paragraph_spacing': False,
 'remove_paragraph_spacing_indent_size': 1.5,
 'renumber_headings': True,
 'replace_scene_breaks': '',
 'search_replace': '[]',
 'series': None,
 'series_index': None,
 'smarten_punctuation': False,
 'sr1_replace': None,
 'sr1_search': None,
 'sr2_replace': None,
 'sr2_search': None,
 'sr3_replace': None,
 'sr3_search': None,
 'start_reading_at': None,
 'subset_embedded_fonts': False,
 'tags': None,
 'timestamp': None,
 'title': None,
 'title_sort': None,
 'toc_filter': None,
 'toc_threshold': 6,
 'toc_title': None,
 'transform_css_rules': '[]',
 'transform_html_rules': '[]',
 'uncompressed_pdf': False,
 'unit': 'inch',
 'unsmarten_punctuation': False,
 'unwrap_factor': 0.45,
 'unwrap_lines': True,
 'use_auto_toc': False,
 'use_profile_size': False,
 'verbose': 2}
DeDRM v10.0.9: Trying to decrypt lnxc8wnj.pdf
DeDRM v10.0.9: lnxc8wnj.pdf is a PDF ebook with encryption EBX_HANDLER
DeDRM v10.0.9: lnxc8wnj.pdf is a PDF ebook (EBX) for UUID 034cb0cc-ea1a-4113-822f-474f8630b186
DeDRM v10.0.9: Looking for new default Adobe Digital Editions Keys after 0.0 seconds
DeDRM v10.0.9: Exception when getting default Adobe Key after 0.0 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 727, in PDFIneptDecrypt
  File "C:\Users\-----\AppData\Roaming\calibre\plugins\DeDRM.zip\adobekey.py", line 293, in adeptkeys
  File "C:\Users\-----\AppData\Roaming\calibre\plugins\DeDRM.zip\adobekey.py", line 273, in CryptUnprotectData
adobekey.ADEPTError: Failed to decrypt user key key (sic)
DeDRM v10.0.9: Finished after 0.0 seconds
InputFormatPlugin: PDF Input running
on C:\Users\----\AppData\Local\Temp\calibre_2fcid2td\lnxc8wnj.pdf
Converting file to html...
Traceback (most recent call last):
  File "runpy.py", line 198, in _run_module_as_main
  File "runpy.py", line 88, in _run_code
  File "site.py", line 95, in <module>
  File "site.py", line 90, in main
  File "site.py", line 50, in run_entry_point
  File "calibre\utils\ipc\worker.py", line 215, in main
  File "calibre\gui2\convert\gui_conversion.py", line 38, in gui_convert_override
  File "calibre\gui2\convert\gui_conversion.py", line 25, in gui_convert
  File "calibre\ebooks\conversion\plumber.py", line 1108, in run
  File "calibre\customize\conversion.py", line 242, in __call__
  File "calibre\ebooks\conversion\plugins\pdf_input.py", line 51, in convert
  File "calibre\ebooks\pdf\pdftohtml.py", line 83, in pdftohtml
calibre.ebooks.ConversionError: pdftohtml failed with return code: 1
Syntax Error: Couldn't find the 'EBX_HANDLER' security handler

I don't know what to do, I don't understand much of it. I looked for similar discussion but it didn't helped. Thank you in advance

**Comment by LucianoCassarini** [comment]
The error you are experiencing appears to be related to the DRM protection of the PDF file.

Make sure the DeDRM plugin is up to date and make sure the DeDRM plugin you are using is compatible with the version of Adobe Digital Editions (ADE) installed on your system.

I also recommend that you try another PDF to rule out that the error is from the PDF you are trying to convert.

**Reply by Lolliponzo** [reply]
I have installed DeDRM 10.0.9. 
How can I know if it is compatible with the versione of Adobe Digital Editions?
I run the debug mode and insert the pdf again, this is the result, maybe it can be helpful.

ApplicationPaletteChange event ignored
Using calibre Qt style: True
calibre Log di debug
calibre 7.6  embedded-python: True
Windows-10-10.0.22631-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.22631')
Python 3.11.5
Windows: ('10', '10.0.22631', 'SP0', 'Multiprocessor Free')
Interface language: it
EXE path: C:\Program Files\Calibre2\calibre-debug.exe
Successfully initialized third party plugins: DeDRM (10, 0, 9) && Obok DeDRM (10, 0, 9)
calibre 7.6  embedded-python: True
Windows-10-10.0.22631-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.22631')
Python 3.11.5
Windows: ('10', '10.0.22631', 'SP0', 'Multiprocessor Free')
Interface language: it
EXE path: C:\Program Files\Calibre2\calibre-debug.exe
Successfully initialized third party plugins: DeDRM (10, 0, 9) && Obok DeDRM (10, 0, 9)
QPA platform: windows
devicePixelRatio: 1.25
logicalDpi: 96.0 x 96.0
physicalDpi: 113.41395348837209 x 113.12164948453608
[0.00] Starting up...
[0.00] Showing splash screen...
[0.16] splash screen shown
[0.16] Initializing db...
[0.16] db initialized
[0.16] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
[0.61] GUI main window shown
[0.66] main UI initialized...
[0.66] Hiding splash screen
DeDRM v10.0.9: Trying to decrypt -----.pdf
DeDRM v10.0.9: -----.pdf is a PDF ebook with encryption EBX_HANDLER
DeDRM v10.0.9: -----.pdf is a PDF ebook (EBX) for UUID 034cb0cc-ea1a-4113-822f-474f8630b186
DeDRM v10.0.9: Looking for new default Adobe Digital Editions Keys after 0.1 seconds
DeDRM v10.0.9: Exception when getting default Adobe Key after 0.1 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 727, in PDFIneptDecrypt
  File "C:\Users\-----\AppData\Roaming\calibre\plugins\DeDRM.zip\adobekey.py", line 293, in adeptkeys
  File "C:\Users\-----\AppData\Roaming\calibre\plugins\DeDRM.zip\adobekey.py", line 273, in CryptUnprotectData
adobekey.ADEPTError: Failed to decrypt user key key (sic)
DeDRM v10.0.9: Finished after 0.2 seconds
Traceback (most recent call last):
  File "calibre\customize\ui.py", line 469, in get_file_type_metadata
  File "calibre\customize\builtins.py", line 324, in get_metadata
  File "calibre\ebooks\metadata\pdf.py", line 124, in get_metadata
ValueError: Could not read info dict from PDF
Syntax Error: Couldn't find the 'EBX_HANDLER' security handler
pdfinfo errored out with return code: 1
[5.95] splash screen hidden
[5.95] Started up in 5.95 seconds with 203 books
stdout+stderr from file dialog helper: [b'', b'']
Added Strategic and Business Planning to db in: 0.1
Added 1 books in 0.8 seconds

**Reply by LucianoCassarini** [reply]
I dont know, the debug output provides information about the DeDRM plugin attempting to decrypt a PDF file with encryption using the EBX_HANDLER security handler. However, it encounters an exception when trying to get the default Adobe Digital Editions keys, resulting in a "Failed to decrypt user key" error.

You can check the official documentation or release notes for the DeDRM plugin. Developers often provide information about compatibility with specific versions of Adobe Digital Editions in these documents.

**Reply by ElleKayEm** [reply]
Did you download the pdf with ADE installed on the same computer and user as calibre/DeDRM?

**Reply by ElleKayEm** [reply]
Also, DeDRM 10.0.9 is compatible with ADE 4.5.

**Reply by Lolliponzo** [reply]
Yes, I have ADE and Calibre on the same computer.

**Reply by ElleKayEm** [reply]
Was ADE authorized with an Adobe ID before you opened the acsm file and downloaded the pdf?

**Reply by Lolliponzo** [reply]
No, it wans't, I downloaded the acsm file and then I installed ADE and authorized it with Adobe ID.

**Reply by ElleKayEm** [reply]
But did you authorize ADE before opening the acsm file?

**Reply by Lolliponzo** [reply]
Yes, I authorized ADE and then I opened the acsm, so it is linked to my account

**Reply by ElleKayEm** [reply, marked as answer]
You could try de-authorizing and re-authorizing ADE with your Adobe ID.  Scroll down on this page to the section under TROUBLESHOOTING REAUTHORIZATION AND DEAUTHORIZATION ISSUES ON WINDOWS: https://helpx.adobe.com/digital-editions/kb/activation-errors-adobe-digital-editions.html

As it says there, be very careful when editing the Windows registry.  Afterward add your book to calibre again.

**Reply by Lolliponzo** [reply]
I did it twice, the first time didn't work but the second one it worked! Finally! Thanks a lot!

**Reply by ElleKayEm** [reply]
Great!  Thanks for reporting back.

**Comment by as5835** [comment]
I purchased DeDRM software. I will help you convert it to PDF for free. Message me

**Reply by ElleKayEm** [reply]
They already solved it with the free software here.
